GoogleForCreators/web-stories-wp
# Embed Block: responsive design improvements

## Summary

- Fix styling of blocks aligned in the center
- Fix styling of blocks aligned left or right
- Ensure blocks scale proportionally depending on available height
- For AMP, uses responsive layout to achieve the same

## Relevant Technical Choices

* CSS is added as inline style as we don't yet have CSS processing in place to have this as an external stylesheet. Plus, it saves an extra network request.

## To-do

**AMP**

* [x] Address AMP validation error for `<amp-story-player>` with `layout="responsive"`  
    Getting the following error:  
    ```
    Tag 'i-amphtml-sizer' is disallowed as child of tag 'amp-story-player'. Child tag must be one of ['a']
    ```
    Blocked by https://github.com/ampproject/amphtml/issues/30414
* [x] Address issue where `<amp-story-player>` has width and height hardcoded.  
    Blocked by https://github.com/ampproject/amphtml/issues/30423

## User-facing changes

Embed block is now properly responsive.

## Testing Instructions

Non-AMP:

1. Insert Web Stories embed block in post, make it quite large
1. View post on small screen sizes, verify that the block shrinks accordingly

For AMP:

1. Activate AMP plugin
1. Switch to nightly channel at https://cdn.ampproject.org/experiments.html
1. View AMP version of the post
1. Verify that the block shrinks accordingly
1. Verify that there are no AMP validation errors for the post.

---



Fixes #3581
